server/server.go

```diff
@@ -85,6 +90,31 @@ func (s *Server) LoadGist(ctx *gin.Context) {
 	ctx.String(resp.StatusCode, string(gistContent))
 }
 
+type versionRequest struct {
+	request
+}
+
+func (s *Server) Version(ctx *gin.Context) {
+	req := versionRequest{}
```

Comment:
I don't see the benefit of using a new struct definition that embeds a `request` if it can be just done
```go
req := request{ctx.Query("server_url")}
cli, err := s.clientForRequest(req)
```

Reply:
I like having separate request structs for different endpoints. It's quite common pattern (you can see it in grpc for example). But if you strongly dislike it, I can change.